rules/integrations/google_workspace/persistence_google_workspace_admin_role_assigned_to_user.toml

```diff
@@ -26,7 +26,52 @@ interval = "10m"
 language = "kuery"
 license = "Elastic License v2"
 name = "Google Workspace Admin Role Assigned to a User"
-note = """## Setup
+note = """## Triage and analysis
+
+### Investigating Google Workspace Admin Role Assigned to a User
+
+Google Workspace roles allow administrators to assign specific permissions to users or groups where the principle of least privilege (PoLP) is recommended. Admin roles in Google Workspace grant users access to the Google Admin console, where further domain-wide settings are accessible. Google Workspace contains prebuilt administrator roles for performing business functions related to users, groups, and services. Custom administrator roles can be created where prebuilt roles are not preferred.
+
+Administrator roles assigned to users will grant them additional permissions and privileges within the Google Workspace domain. Administrative roles also give users access to the admin console, where domain-wide settings can be adjusted. Threat actors may rely on these new privileges to thus further their intrusion efforts and laterally move throughout the organization. Users with unexpected administrative privileges may also cause operational disfunction if unfamiliar settings are adjusted without warning.
+
+This rule identifies when a Google Workspace administrative role is assigned to a user.
+
+#### Possible investigation steps
+
+- Identify the user account(s) associated by reviewing `user.name` or `user.email` in the alert.
+  - The `user.target.email` field contains the user who received the admin role.
+- Identify the role given to the user by reviewing `google_workspace.admin.role.name` in the alert.
+- With the user identified, verify administrative privileges are scoped properly.
+- To identify other users with this role, search for `event.action: ASSIGN_ROLE`.
+  - Add `google_workspace.admin.role.name` with the role added as an additional filter.
```

Comment:
This is suggesting to apply this search for field `event.action` and value `ASSIGN_ROLE` in elasticsearch. It will help identify other users who were recently given new roles. This can be indicative of additional compromised accounts and privilege escalation.